src/Lib/Lionk.Core/Model/Component/Cyclic/CyclicExecutorService.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright © 2024 Lionk Project
+
+using System.Timers;
+
+using Lionk.Core.Component;
+using Lionk.Log;
+
+namespace Lionk.Core.Model.Component.Cyclic;
+
+/// <summary>
+/// This class is used to execute cyclic components.
+/// </summary>
+public class CyclicExecutorService : ICyclicExecutorService
+{
+    private readonly CancellationTokenSource _cancellationTokenSource = new();
+    private readonly System.Timers.Timer _timer;
+    private Thread _thread;
+
+    private void Execute()
+    {
+        try
+        {
+            while (!_cancellationTokenSource.Token.IsCancellationRequested)
+            {
+                switch (State)
+                {
+                    case CycleState.Paused:
+                        continue;
+                    case CycleState.Running:
+                        Running();
+                        break;
+                    case CycleState.Stopped:
+                        Stopping();
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
+        }
+        catch (ThreadInterruptedException)
+        {
+            State = CycleState.Aborted;
+        }
+    }
+
+    private void Stopping()
+    {
+        State = CycleState.Stopping;
+        _cancellationTokenSource.Cancel();
+    }
+
+    private void Running()
+    {
+        foreach (ICyclicComponent component in Components)
+        {
+            DateTime now = DateTime.UtcNow;
+            if (component.NextExecution > now) continue;
+            _timer.Start();
+            component.Execute();
+            _timer.Stop();
+            component.NbCycle++;
+        }
+    }
+
+    /// <inheritdoc/>
+    public CycleState State { get; private set; }
+
+    /// <inheritdoc/>
+    public TimeSpan WatchDogTime { get; private set; }
+
+    /// <inheritdoc/>
+    public IEnumerable<ICyclicComponent> Components => ComponentService.GetInstancesOfType<ICyclicComponent>();
+
+    /// <inheritdoc/>
+    public IComponentService ComponentService { get; private set; }
+
+    /// <inheritdoc/>
+    public void Start()
+    {
+        foreach (ICyclicComponent component in Components)
+        {
+            component.StartedDate = DateTime.UtcNow;
+            component.NbCycle = 0;
+        }
+
+        if (State == CycleState.Running) return;
```

Comment:
This control should be on top of the foreach no ?

Reply:
It's not the purpose of this if, it just avoids to start twice the thread. But reanalyzing this method, I realize that we have a problem with new Cycle components added on the fly.
- The `IEnumerable<Component> Component` can't be modified when it's used. we need to manage concurrency
- New CyclicComponents won't be properly initialized. 
We need to review the workflow of this class